web/react/utils/utils.jsx

```diff
@@ -437,60 +528,86 @@ module.exports.textToJsx = function(text, options) {
+                }
+                else
+                    inner.push(<span key={word+i+z+"_span"}>{prefix}<a key={word+i+z+"_hash"} className={"theme " + mClass + highlightSearchClass} href="#" onClick={function(value) { return function() { module.exports.searchForTerm(value); } }(trimWord)}>{trimWord}</a>{suffix} </span>);
 
             } else if (implicitKeywords.indexOf(trimWord) !== -1 || implicitKeywords.indexOf(trimWord.toLowerCase()) !== -1) {
-                var suffix = word.match(puncEndRegex);
-                var prefix = word.match(puncStartRegex);
-
                 if (trimWord.charAt(0) === '@') {
                     if (searchTerm === trimWord.substring(1).toLowerCase()) {
                         highlightSearchClass = " search-highlight";
                     }
-                    inner.push(<span key={word+i+z+"_span"} key={name+i+z+"_span"}>{prefix}<a className={mentionClass + highlightSearchClass} key={name+i+z+"_link"} href="#">{trimWord}</a>{suffix} </span>);
+                    if (useMarkdown) {
+                        prefixSpan ? inner.push(prefixSpan) : null;
+                        inner.push(<span key={word+i+z+"_span"} key={name+i+z+"_span"}><a className={mentionClass + highlightSearchClass} key={name+i+z+"_link"} href="#" dangerouslySetInnerHTML={{__html: trimWord}} /></span>);
+                        suffixSpan ? inner.push(suffixSpan) : null;
+                    }
+                    else
+                        inner.push(<span key={word+i+z+"_span"} key={name+i+z+"_span"}>{prefix}<a className={mentionClass + highlightSearchClass} key={name+i+z+"_link"} href="#">{trimWord}</a>{suffix} </span>);
                 } else {
-                    inner.push(<span key={word+i+z+"_span"}>{prefix}<span className={mentionClass + highlightSearchClass}>{module.exports.replaceHtmlEntities(trimWord)}</span>{suffix} </span>);
+                    if (useMarkdown) {
+                        prefixSpan ? inner.push(prefixSpan) : null;
+                        inner.push(<span key={word+i+z+"_span"}><span className={mentionClass + highlightSearchClass} dangerouslySetInnerHTML={{__html: trimWord}} /></span>);
+                        suffixSpan ? inner.push(suffixSpan) : null;
+                    }
+                    else
+                        inner.push(<span key={word+i+z+"_span"}>{prefix}<span className={mentionClass + highlightSearchClass}>{module.exports.replaceHtmlEntities(trimWord)}</span>{suffix} </span>);
                 }
 
             } else if (word === "") {
                 // if word is empty dont include a span
             } else {
-                inner.push(<span key={word+i+z+"_span"}><span className={highlightSearchClass}>{module.exports.replaceHtmlEntities(word)}</span> </span>);
+                if (useMarkdown)
+                    inner.push(<span key={word+i+z+"_span"}><span className={highlightSearchClass} dangerouslySetInnerHTML={{__html: word}} /> </span>);
```

Comment:
Like my comment on line 511, this could lead to weird nesting of tags if marked were to emit something where there is any whitespace between the start and end tags.